src/sentry/services/hybrid_cloud/auth/impl.py

```diff
@@ -114,3 +168,76 @@ def get_org_ids_with_scim(
                 flags=F("flags").bitor(AuthProvider.flags.scim_enabled)
             ).values_list("organization_id", flat=True)
         )
+
+
+class FakeRequestDict:
+    d: Mapping[str, str | bytes | None]
+
+    def __init__(self, **d: Any):
+        self.d = d
+
+    def __getitem__(self, item: str) -> str | bytes:
+        result = self.d[item]
+        if result is None:
+            raise KeyError(f"Key '{item}' does not exist")
```

Comment:
Nit: `f"Key {item!r} does not exist"` adds the quotation marks for you and handles escapes, if necessary.